src/Google.Api.Gax/Api/Gax/ServiceEndpointSettings.cs

```diff
@@ -60,5 +61,19 @@ public string Host
             Host = Host,
             Port = Port,
         };
+
+        /// <summary>
+        /// Creates a channel for the host and port in this object, with defaults for unspecified values.
+        /// </summary>
+        /// <param name="defaultHost">The host to use if <see cref="Host"/> is null. Must not be null.</param>
+        /// <param name="defaultPort">The port to use if <see cref="Port"/> is null.</param>
+        /// <param name="credentials">The credentials to use with the channel. Must not be null.</param>
+        /// <returns></returns>
+        public Channel CreateChannel(string defaultHost, int defaultPort, ChannelCredentials credentials)
```

Comment:
Should this method be in the settings class? This puts functionality into a class that currently just expresses configuration.
I'd rather see this as a static method in ClientHelper.

As a user having this public method here might be confusing. ServiceEndpointSettings is a class that users are expected to directly interact with (via a concrete service-specific settings class, inherited from this), and having this public method here may well be confusing, not least because they might think this they are meant to call themselves to get a channel.
This method is purely for our internal use, so I think it's better suited on ClientHelper, which is also purely for our internal use.
(and on ServiceSettingsBase.GetDeadlineCalculator)